src/collision.cpp

```diff
@@ -446,48 +447,48 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 		// Avoid infinite loop
 		loopcount++;
 		if (loopcount >= 100) {
-			warningstream << "collisionMoveSimple: Loop count exceeded, aborting to avoid infiniite loop" << std::endl;
+			warningstream << "collisionMoveSimple: Loop count exceeded, aborting to avoid infinite loop" << std::endl;
 			break;
 		}
 
 		aabb3f movingbox = box_0;
 		movingbox.MinEdge += *pos_f;
 		movingbox.MaxEdge += *pos_f;
 
-		CollisionAxis nearest_collided = COLLISION_AXIS_NONE;
 		f32 nearest_dtime = dtime;
-		int nearest_boxindex = -1;
+		NearbyCollisionInfo *nearest_info = nullptr;
 
 		/*
 			Go through every nodebox, find nearest collision
 		*/
-		for (u32 boxindex = 0; boxindex < cinfo.size(); boxindex++) {
-			const NearbyCollisionInfo &box_info = cinfo[boxindex];
+		for (auto &box_info : cinfo) {
```

Comment:
Simplifying this code here: Rather than storing the index and re-indexing a variable we already have, just store the pointer we already have.

Unfortunately, `nearest_info` cannot be a reference since it is nullable (See the null check below)